tasks/ModelTask.cpp

```diff
@@ -222,20 +222,37 @@ void ModelTask::updateLinks(base::Time const& time)
     for(ExportedLinks::const_iterator it = exported_links.begin(); it != exported_links.end(); ++it)
     {
         math::Pose source2world = math::Pose::Zero;
+        math::Vector3 sourceInWorld_linear_vel  = math::Vector3::Zero;
+        math::Vector3 sourceInWorld_angular_vel = math::Vector3::Zero;
         if (it->second.source_link_ptr)
+        {
             source2world = it->second.source_link_ptr->GetWorldPose();
+            sourceInWorld_linear_vel  = it->second.source_link_ptr->GetWorldLinearVel();
+            sourceInWorld_angular_vel = it->second.source_link_ptr->GetWorldAngularVel();
+        }
+
         math::Pose target2world = math::Pose::Zero;
         if (it->second.target_link_ptr)
             target2world = it->second.target_link_ptr->GetWorldPose();
+
         math::Pose source2target( math::Pose(source2world - target2world) );
+        math::Vector3 sourceInTarget_angular_vel(target2world.rot.RotateVectorReverse(sourceInWorld_angular_vel));
+        math::Vector3 sourceInTarget_linear_vel (target2world.rot.RotateVectorReverse(sourceInWorld_linear_vel));
```

Comment:
Alright.

I just have doubt about the use of sourceFrame==targetFrame.
It means that we interpret the velocities as absolute (velocities of the sourceFrame related to a inertial frame, not the targetFrame) and represent them in targetFrame (for linear case) and sourceFrame (angular case). 

The relative velocities of the sourceFrame related to the targetFrame should be zero if they are the same.